src/Settings/Flavours/Quickest.hs

```diff
@@ -2,13 +2,16 @@ module Settings.Flavours.Quickest (quickestFlavour) where
 
 import Flavour
 import Predicate
+import Oracles.Config.Flag (platformSupportsSharedLibs)
 import {-# SOURCE #-} Settings.Default
 
 quickestFlavour :: Flavour
 quickestFlavour = defaultFlavour
     { name        = "quickest"
     , args        = defaultBuilderArgs <> quickestArgs <> defaultPackageArgs
-    , libraryWays = append [vanilla]
+    , libraryWays = mconcat
+                    [ append [vanilla]
+                    , notStage0 ? platformSupportsSharedLibs ? append [dynamic] ]
```

Comment:
The `quickest` flavour is supposed to be the quickest one that produces a working GHC, to be used by CI bots and avoid time limits, so I suggest we stick to pure `vanilla`.